classes/submission/SubmissionFileDAODelegate.inc.php

```diff
@@ -359,6 +362,18 @@ function _deleteDependentObjects($submissionFile) {
 				ASSOC_TYPE_SUBMISSION_FILE, $submissionFile->getFileIdAndRevision()
 		);
 	}
+	
+	/**
+	 * @copydoc DAO::getAdditionalFieldNames()
+	 */
+	function getAdditionalFieldNames() {
```

Comment:
I think I'd prefer to use a new column in the relevant tables rather than storing this as a setting. It will be a core feature, and there are likely to be performance implications that require some optimization, and that'll be easier if it's a full-fledged column.